public/app/features/trails/helpers/MetricDatasourceHelper.ts

```diff
@@ -103,19 +113,33 @@ export class MetricDatasourceHelper {
   }
 
   /**
-   *
-   * If a metric name + _bucket exists in the classic histograms, then it is a native histogram
+   * Identify native histograms by 2 strategies.
+   * 1. querying classic histograms and all metrics,
+   * then comparing the results and build the collection of native histograms.
+   * 2. querying all metrics and checking if the metric is a histogram type and dies not have the bucket suffix.
    *
    * classic histogram = test_metric_bucket
    * native histogram = test_metric
+   *
    * @param metric
-   * @returns
+   * @returns boolean
    */
   public isNativeHistogram(metric: string): boolean {
     if (!metric) {
       return false;
     }
 
+    // check when fully migrated, we only have metadata, and there are no more classic histograms
+    const metadata = this._metricsMetadata;
+    // suffix is not 'bucket' and type is histogram
+    const suffix: string = metric.split('_').pop() ?? '';
+    // the string is not equal to bucket
+    const notClassic = suffix !== 'bucket';
+    if (metadata && metadata[metric] && metadata[metric].type === 'histogram' && notClassic) {
```

Comment:
nit: this could be tidier with optional chaining

```suggestion
    if (metadata?.[metric]?.type === 'histogram' && notClassic) {
```